### PR TITLE
[WIP] Add Import Service and UploadSST RPC

### PR DESCRIPTION
**Pull request by huachaohuang**
Allow a client to upload an SST file to TiKV.

### CODE REVIEW

File: src/server/service/import.rs [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+use futures::{Future, Stream};
+use futures_cpupool::{Builder, CpuPool};
+use kvproto::importpb::*;
+use kvproto::importpb_grpc;
+
+use server::metrics::*;
+use server::errors::Error;
+use raftstore::store::{UploadDir, Uploader};
+
+macro_rules! send_rpc_response {
```
Comment (siddontang) [inline src/server/service/import.rs]: seem not necessary to define the macro here, we only use once.

File: src/raftstore/store/upload.rs [hunk 2]
```diff
@@ -0,0 +1,390 @@
...
+        "{}_{}_{}_{}_{}.sst",
+        uuid.simple().to_string(),
+        h.get_cfname(),
+        h.get_region_id(),
+        h.get_region_epoch().get_conf_ver(),
+        h.get_region_epoch().get_version(),
+    )))
+}
+
+fn file_exists<P: AsRef<Path>>(path: P) -> Error {
```
Comment (siddontang) [inline src/raftstore/store/upload.rs]: seem we should move these common functions to util mod.

Comment (siddontang) [reply src/raftstore/store/upload.rs]: my fault, these functions are for error, I think we should name `file_exists_error`

File: src/server/service/import.rs (hunk 1 above)
Comment (huachaohuang) [reply src/server/service/import.rs]: It will be used in other RPCs later.

Comment (siddontang) [reply src/server/service/import.rs]: Like other services, we should avoid macro here.

**Comment by huachaohuang** [comment]
Moved to another place now.

**Pull request closed without merging**